anni-common/src/models.rs

```diff
@@ -38,13 +44,59 @@ impl<'a> Clone for RawTrackIdentifier<'a> {
     }
 }
 
+impl<'a> Display for RawTrackIdentifier<'a> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}/{}/{}", self.album_id, self.disc_id, self.track_id)
+    }
+}
+
 #[derive(Hash, PartialEq, Eq)]
 pub struct TrackIdentifier {
-    inner: RawTrackIdentifier<'static>,
+    pub inner: RawTrackIdentifier<'static>,
 }
 
 impl<'a> Borrow<RawTrackIdentifier<'a>> for TrackIdentifier {
     fn borrow(&self) -> &RawTrackIdentifier<'a> {
         &self.inner
     }
 }
+
+impl FromStr for TrackIdentifier {
+    type Err = ParseError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let read_u8 =
+            |s: &mut Split<_>| s.next().ok_or(ParseError)?.parse().map_err(|_| ParseError);
+
+        let mut sp = s.split('/');
+
+        let album_id = sp.next().ok_or(ParseError)?;
+        let disc_id = read_u8(&mut sp)?;
+        let track_id = read_u8(&mut sp)?;
+
+        Ok(RawTrackIdentifier::new(album_id, disc_id, track_id).to_owned())
+    }
+}
+
+impl Display for TrackIdentifier {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        self.inner.fmt(f)
+    }
+}
+
+impl Clone for TrackIdentifier {
+    fn clone(&self) -> Self {
+        self.inner.to_owned()
+    }
+}
+
+#[derive(Debug, Clone, Copy)]
+pub struct ParseError;
```

Comment:
Use `thiserror` instead?

Reply:
Is it too heavy since it is just a trivial error type?

Reply:
We may separate the error to somewhat like `invalid album id`, `invalid disc id` and `invalid track id`

Reply:
updated